### PR TITLE
jenkins_job_info: Remove necessities of password or token.

### PR DESCRIPTION
**Pull request by unnecessary-username**
##### SUMMARY

Remove necessities of password or token in the jenkins_job_info module.

There are many use cases that Jenkins is allowed to visit without authentication. Also in the original implementation, the get_jenkins_connection function supports connections that exclude of password or token. So I think it should not have the restriction to make password or token necessary.

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

jenkins_job_info

### CODE REVIEW

File: changelogs/fragments/2948-jenkins_job_info-remove_necessities_on_password_or_token.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - jenkins_job_info - Remove necessities on password or token (https://github.com/ansible-collections/community.general/pull/2948).
```
Comment (aminvakil) [inline changelogs/fragments/2948-jenkins_job_info-remove_necessities_on_password_or_token.yml:2]:
```suggestion
  - jenkins_job_info - remove necessities on password or token (https://github.com/ansible-collections/community.general/pull/2948).
```

**Review by aminvakil — CHANGES_REQUESTED** [review]
LGTM
I'm just blocking this from getting merged until we decide whether we want to release 3.3.2 or not.

**Comment by felixfontein** [comment]
@aminvakil there's no need to block a PR for merging to `main`. It's the backport PRs that must not be merged before that's decided :)

File: changelogs/fragments/2948-jenkins_job_info-remove_necessities_on_password_or_token.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - jenkins_job_info - remove necessities on password or token (https://github.com/ansible-collections/community.general/pull/2948).
```
Comment (felixfontein) [inline changelogs/fragments/2948-jenkins_job_info-remove_necessities_on_password_or_token.yml:2]:
```suggestion
  - jenkins_job_info - the ``password`` and ``token`` parameters can also be omitted to retrieve only public information (https://github.com/ansible-collections/community.general/pull/2948).
```

File: plugins/modules/web_infrastructure/jenkins_job_info.py [hunk 3]
```diff
@@ -33,12 +33,10 @@
     type: str
     description:
       - Password to authenticate with the Jenkins server.
-      - This is a required parameter, if C(token) is not provided.
```
Comment (felixfontein) [inline plugins/modules/web_infrastructure/jenkins_job_info.py:36]: Can you replace this with `This is mutually exclusive with C(token).`? (That should have already been mentioned there before IMO.) Same for the other.

File: changelogs/fragments/2948-jenkins_job_info-remove_necessities_on_password_or_token.yml (hunk 2 above)
Comment (unnecessary-username) [reply changelogs/fragments/2948-jenkins_job_info-remove_necessities_on_password_or_token.yml]: Done in the latest commit.

File: plugins/modules/web_infrastructure/jenkins_job_info.py (hunk 3 above)
Comment (unnecessary-username) [reply plugins/modules/web_infrastructure/jenkins_job_info.py:36]: Done in the latest commit.

File: plugins/modules/web_infrastructure/jenkins_job_info.py [hunk 4]
```diff
@@ -33,12 +33,12 @@
...
+      - This is mutually exclusive with C(token).
```
Comment (felixfontein) [inline plugins/modules/web_infrastructure/jenkins_job_info.py:36]:
```suggestion
      - This is mutually exclusive with I(token).
```

File: plugins/modules/web_infrastructure/jenkins_job_info.py [hunk 5]
```diff
@@ -33,12 +33,12 @@
...
   token:
     type: str
     description:
       - API token used to authenticate with the Jenkins server.
-      - This is a required parameter, if C(password) is not provided.
+      - This is mutually exclusive with C(password).
```
Comment (felixfontein) [inline plugins/modules/web_infrastructure/jenkins_job_info.py:41]:
```suggestion
      - This is mutually exclusive with I(password).
```

**Review by felixfontein — COMMENTED** [review]
Sorry for not spotting that earlier, the old documentation was using the wrong formatting for option names...

**Review by felixfontein — APPROVED** [review]

**Review by aminvakil — APPROVED** [review]
LGTM

**Comment by felixfontein** [comment]
@unnecessary-username thanks for implementing this!
@aminvakil thanks for reviewing!

**Pull request merged**